### ISSUE TITLE
Arch misdetection in Linux 32bits

### ISSUE THREAD

**Issue by memsharded** [conan-io/conan#323]
So it appears in conan.conf:

arch=i686

Which is not a recognized architecture in settings, it should be x86. Reported by @coding3d

**Comment by hoxnox** [comment]
Now it doesn't recognizes on Debian i686 jessie. It needs i686.

**Comment by memsharded** [comment]
Don't understand. The default settings only have values for x86 and x86_64 (basically 32 and 64 bits). It seems to be fine with those values. Are you using the i686 setting in your own setting? How would you handle it in your builds?

**Comment by hoxnox** [comment]
I have to manualy correct ~/.conan/conan.conf and set arch=x86 here. If I launch `conan install` on Debian-x86 jessie, I constantly have errors:

```sh
vdebian% conan --version
Conan version 0.9.4
vdebian% rm -r ~/.conan 
vdebian% conan install

It seems to be the first time you run conan
Auto detecting your dev setup to initialize conan.conf
Found gcc 4.9
Default conan.conf settings
	os=Linux
	arch=i686
	compiler=gcc
	compiler.version=4.9
	compiler.libcxx=libstdc++
	build_type=Release
*** You can change them in ~/.conan/conan.conf ***
*** Or override with -s compiler='other' -s ...s***


ERROR: 'i686' is not a valid 'settings.arch' value.
Possible values are ['armv6', 'armv7', 'armv7hf', 'armv8', 'x86', 'x86_64']
vdebian% rm -r ~/.conan 
vdebian% conan install -s arch='x86' -s compiler=gcc

It seems to be the first time you run conan
Auto detecting your dev setup to initialize conan.conf
Found gcc 4.9
Default conan.conf settings
	os=Linux
	arch=i686
	compiler=gcc
	compiler.version=4.9
	compiler.libcxx=libstdc++
	build_type=Release
*** You can change them in ~/.conan/conan.conf ***
*** Or override with -s compiler='other' -s ...s***

ERROR: 'i686' is not a valid 'settings.arch' value.
Possible values are ['armv6', 'armv7', 'armv7hf', 'armv8', 'x86', 'x86_64']
vdebian% conan install -s arch='x86' -s compiler=gcc
ERROR: 'i686' is not a valid 'settings.arch' value.
Possible values are ['armv6', 'armv7', 'armv7hf', 'armv8', 'x86', 'x86_64']

```

**Comment by hoxnox** [comment]
Oh. This patch is absent in 0.9.4. I think it'll fix my problem. Sorry.

**Comment by memsharded** [comment]
Yes, we add the label "fixed" to say that it is implemented, and merged in develop branch. We close the issue when it is released (the milestone is 0.10, will be the next release), precisely for this reason. No worries at all! Better ask or report a false positive than not reporting a true positive. Thank you!

**Comment by lasote** [comment]
Fixed in 0.10.0, please try it and reopen if needed.
